pkg/controller/cainjector/indexers.go

```diff
@@ -68,82 +89,52 @@ func buildCertToInjectableFunc(listTyp runtime.Object, resourceName string) cert
 	}
 }
 
-// secretForCertificateMapper is a Mapper that converts secrets up to injectables, through certificates.
-type secretForCertificateMapper struct {
-	Client                  client.Reader
-	log                     logr.Logger
-	certificateToInjectable certificateToInjectableFunc
-}
-
-func (m *secretForCertificateMapper) Map(obj client.Object) []ctrl.Request {
-	// grab the certificate, if it exists
-	certName := owningCertForSecret(obj.(*corev1.Secret))
-	if certName == nil {
-		return nil
-	}
-
-	secretName := types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}
-	log := m.log.WithValues("secret", secretName, "certificate", *certName)
-
-	var cert cmapi.Certificate
-	// confirm that a service owns this cert
-	if err := m.Client.Get(context.Background(), *certName, &cert); err != nil {
-		// TODO(directxman12): check for not found error?
-		log.Error(err, "unable to fetch certificate that owns the secret")
-		return nil
-	}
-
-	return m.certificateToInjectable(log, m.Client, *certName)
-}
-
-// certMapper is a mapper that converts Certificates up to injectables
-type certMapper struct {
-	Client       client.Reader
-	log          logr.Logger
-	toInjectable certificateToInjectableFunc
-}
-
-func (m *certMapper) Map(obj client.Object) []ctrl.Request {
-	certName := types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}
-	log := m.log.WithValues("certificate", certName)
-	return m.toInjectable(log, m.Client, certName)
-}
-
-var (
-	// injectFromPath is the index key used to look up the value of inject-ca-from on targeted objects
-	injectFromPath = ".metadata.annotations.inject-ca-from"
-)
+// certFromSecretToInjectableMapFuncBuilder returns a handler.MapFunc that, for
+// a Certificate change, ensures that if this Certificate that is configured as
+// a CA source for an injectable via inject-ca-from annotation, a reconcile loop
+// will be triggered for this injectable
+func certToInjectableMapFuncBuilder(cl client.Reader, log logr.Logger, config setup) handler.MapFunc {
+	return func(ctx context.Context, obj client.Object) []ctrl.Request {
+		certName := types.NamespacedName{Namespace: obj.GetNamespace(), Name: obj.GetName()}
+		log = log.WithValues("type", config.resourceName, "certificate", certName)
+		objs := config.listType.DeepCopyObject().(client.ObjectList)
+		if err := cl.List(context.Background(), objs, client.MatchingFields{injectFromPath: certName.String()}); err != nil {
+			log.Error(err, "unable to fetch injectables associated with certificate")
+			return nil
+		}
 
-// injectableCAFromIndexer is an IndexerFunc indexing on certificates
-// referenced by injectables.
-func injectableCAFromIndexer(rawObj client.Object) []string {
```

Comment:
Moved the helper/util funcs to the bottom and the three event handler funcs to the top. Perhaps the other functions could have been put into a separate file, but for not I think this is ok